scripts/add_connector.sh

```diff
@@ -1,33 +1,60 @@
 #! /usr/bin/env bash
-
+function find_prev_connector() {
+    self=scripts/add_connector.sh
+    cp $self $self.tmp
+    connectors=(aci adyen airwallex applepay authorizedotnet bambora bluesnap braintree checkout cybersource dlocal fiserv forte forte forte forte forte forte forte forte forte forte forte forte forte forte globalpay klarna mollie multisafepay nuvei payu rapyd shift4 stripe trustpay utils worldline worldpay "$1")
+    IFS=$'\n' sorted=($(sort <<<"${connectors[*]}")); unset IFS
+    res=`echo ${sorted[@]}`
+    sed -i'' -e "s/^    connectors=.*/    connectors=($res \"\$1\")/" $self.tmp
+    for i in "${!sorted[@]}"; do
+    if [ "${sorted[$i]}" = "$1" ] && [ $i != "0" ]; then
+        eval "$2='${sorted[i-1]}'"
+        mv $self.tmp $self
+        rm $self.tmp-e
+        return 0
+    fi
+    done
+    mv $self.tmp $self
+    rm $self.tmp-e
+    eval "$2='aci'"
```

Comment:
What is exactly happening here? It'd be preferable if this had comments.

Reply:
we are trying to find the previous connector so that we can add our current connector next to it. this helps in keeping the connectors in sorted order

Reply:
`Aci` is for corner case, if there is no connector with name `ab*` it will be added after Aci.